File: 1693-sum-of-all-odd-length-subarrays/solution.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {
        int n=arr.length;

        int sum=0;

        for(int i=0;i<n;i++){
            int tsa=(i+1)*(n-i);
            int osa=(tsa+1)/2;
            sum+=arr[i]*osa;
        }
        return sum;
    }
}

File: 1693-sum-of-all-odd-length-subarrays/solution.py
class Solution:
    def sumOddLengthSubarrays(self, arr: List[int]) -> int:
        tot=0
        for i in range(len(arr)):
            for j in range(i,len(arr),2):
                tot+=sum(arr[i:j+1])
        return tot
        
